### PR TITLE
Fix for #3722

### PR DESCRIPTION
**Pull request by ehiggs**
When using init, add [[bin]] and [lib] section where appropriate.

### CODE REVIEW

File: tests/init.rs [hunk 1]
```diff
@@ -94,36 +94,48 @@ fn bin_already_exists(explicit: bool, rellocation: &str) {
     let mut new_content = Vec::new();
     File::open(&sourcefile_path).unwrap().read_to_end(&mut new_content).unwrap();
     assert_eq!(Vec::from(content as &[u8]), new_content);
+
+	let mut cargo_content = String::new();
```
Comment (alexcrichton) [inline tests/init.rs]: The indentation here looks like it may be a bit off?

**Comment by alexcrichton** [comment]
Looks great to me, thanks!

**Comment by matklad** [comment]
@alexcrichton look's like you've forgotten to r+? Or is there anything else here to do?

**Comment by ehiggs** [comment]
Further issues #3722: if `cargo init` detects a main function, but wasn't given `--bin` then it still prints `Created library project`. But that can be done in a separate PR imho.

**Pull request merged**